clickhouse/columns/string.cpp

```diff
@@ -42,104 +70,187 @@ void ColumnFixedString::Append(ColumnRef column) {
 }
 
 bool ColumnFixedString::Load(CodedInputStream* input, size_t rows) {
-    data_.reserve(data_.size() + rows);
-
-    for (size_t i = 0; i < rows; ++i) {
-        std::string s;
-        s.resize(string_size_);
-
-        if (!WireFormat::ReadBytes(input, &s[0], s.size())) {
-            return false;
-        }
-
-        data_.push_back(std::move(s));
+    data_.resize(string_size_ * rows);
```

Comment:
unfortunately there is no easy way (not even hard way) to avoid pre-filling std::string, and that is why I use `Block` for `ColumnString` below.

As for assumptions: somehow those assumptions are opposite for rest of column types, which erase previous data. So I had to unify behaviors.

Reply:
Right, but here, that zero-filling still takes place.

If not now, maybe in future, this could be used. Looks like the header itself is standalone:
https://github.com/facebook/folly/blob/master/folly/memory/UninitializedMemoryHacks.h

Reply:
I guess this is not a bog problem for `FixedString`, but yes, we might want to use Block here too/ That facebook hack...